bootstrap/config/environment.go

```diff
@@ -116,6 +127,58 @@ func (e *Environment) OverrideConfiguration(lc logger.LoggingClient, serviceConf
 	return overrideCount, nil
 }
 
+// buildPaths create the path strings for all settings in the Config tree's key map
+func (e *Environment) buildPaths(keyMap map[string]interface{}) []string {
+	var paths []string
+
+	for key, item := range keyMap {
+		if reflect.TypeOf(item).Kind() != reflect.Map {
+			paths = append(paths, key)
+			continue
+		}
+
+		subMap := item.(map[string]interface{})
+
+		subPaths := e.buildPaths(subMap)
+		for _, path := range subPaths {
+			paths = append(paths, fmt.Sprintf("%s.%s", key, path))
+		}
+	}
+
+	return paths
+}
+
+// buildUppercasePathMap builds a map where the key is the uppercase version of the path
+// and the value is original version of the path
+func (e *Environment) buildUppercasePathMap(paths []string) map[string]string {
+	ucMap := make(map[string]string)
+	for _, path := range paths {
+		ucMap[strings.ToUpper(path)] = path
+	}
+
+	return ucMap
+}
+
+// getKeyForMatchedPath searches for match of the environment variable name with the uppercase path (pathMap keys)
+// If matched found to original path (pathMap values) is returned as the "key"
+// For backward compatibility a case insensitive comparision is currently used.
+// TODO: For release v2.0.0 Change this to NOT check that `envVarName` is uppercase and only compare against uppercase
```

Comment:
Should we put this as an issue to address in the future?

Reply:
Done https://github.com/edgexfoundry/go-mod-bootstrap/issues/71